### PR TITLE
Drop regex for canonicalization

### PR DESCRIPTION
**Pull request by wussler**
Change canonicalization to `strings.TrimRight` instead of regex

### CODE REVIEW

**Review by twiss — APPROVED** [review]

File: helper/cleartext_test.go [hunk 1]
```diff
@@ -45,11 +45,5 @@ func TestSignClearText(t *testing.T) {
 	if err != nil {
 		t.Fatal("Cannot parse message:", err)
 	}
-	assert.Exactly(t, canonicalizeAndTrim(inputPlainText), string(clearTextMessage.GetBinary()))
-}
-
-func canonicalizeAndTrim(text string) string {
```
Comment (henrybear327) [inline helper/cleartext_test.go:51]: I think the refactored code change is not the same as the original regex. 

In the original implementation, only the trailing space of the entire string input is trimmed, and then all the newlines are globally replaced with \r\n.

But, in the refactored implementation, spaces are trimmed on every line.

File: crypto/message.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ func NewPlainMessageFromFile(data []byte, filename string, time uint32) *PlainMe
 // ready for encryption, signature, or verification from an unencrypted string.
 func NewPlainMessageFromString(text string) *PlainMessage {
 	return &PlainMessage{
-		Data:     []byte(strings.ReplaceAll(strings.ReplaceAll(text, "\r\n", "\n"), "\n", "\r\n")),
+		Data:     []byte(internal.CanonicalizeAndTrim(text)),
```
Comment (zhj4478) [inline crypto/message.go:91]: we shouldn't trim all inputs. we should only trim when we need a trim. I think only clear-text signatures need to trim.

File: helper/cleartext_test.go (hunk 1 above)
Comment (wussler) [reply helper/cleartext_test.go:51]: This is intended behaviour according to RFC 4880 paragraphs 5.2.1 and 7.1

File: crypto/message.go (hunk 2 above)
Comment (wussler) [reply crypto/message.go:91]: On the BE we convert some instances of signed messages to cleartext according to the preferences, so this is as of now needed for all messages. Please refer to https://github.com/ProtonMail/gopenpgp/pull/136

**Pull request merged**